pkg/master/controller.go

```diff
@@ -257,18 +260,20 @@ type EndpointReconciler interface {
 // masterCountEndpointReconciler reconciles endpoints based on a specified expected number of
 // masters. masterCountEndpointReconciler implements EndpointReconciler.
 type masterCountEndpointReconciler struct {
-	masterCount      int
-	endpointRegistry endpoint.Registry
+	masterCount       int
+	endpointRegistry  endpoint.Registry
+	configmapRegistry configmap.Registry
```

Comment:
Kubelet reads the configmap and writes it to disk on updates. So changes should be made to the configmap in apiserver, kubelet reads it, then apiserver observes the changed file (via inotify) and can decide to restart (should do this to begin with) or just change the flags in place if possible. The latter bit is harder and should probably come in a subsequent PR.